src/lib.rs

```diff
@@ -18,87 +22,96 @@ mod tokenizer;
 #[cfg(feature = "debug")]
 mod debug;
 
-/// Formats whitespace in a SQL string to make it easier to read.
-/// Optionally replaces parameter placeholders with `params`.
-pub fn format(query: &str, params: &QueryParams, options: &FormatOptions) -> String {
-    let named_placeholders = matches!(params, QueryParams::Named(_));
-
-    let tokens = tokenizer::tokenize(query, named_placeholders, options);
-    formatter::format(&tokens, params, options)
-}
-
 /// The SQL dialect to use. This affects parsing of special characters.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
 pub enum Dialect {
     /// Generic SQL syntax, most dialect-specific constructs are disabled
+    #[default]
     Generic,
     /// Enables array syntax (`[`, `]`) and operators
     PostgreSql,
     /// Enables `[bracketed identifiers]` and `@variables`
     SQLServer,
 }
 
-/// Options for controlling how the library formats SQL
-#[derive(Debug, Clone)]
+/// SQL FormatOptions
+#[derive(Debug, Clone, Builder)]
 pub struct FormatOptions<'a> {
     /// Controls the type and length of indentation to use
     ///
-    /// Default: 2 spaces
-    pub indent: Indent,
+    #[builder(default, into)]
+    indent: Indent,
     /// When set, changes reserved keywords to ALL CAPS
-    ///
-    /// Default: false
-    pub uppercase: Option<bool>,
+    uppercase: Option<bool>,
     /// Controls the number of line breaks after a query
-    ///
-    /// Default: 1
-    pub lines_between_queries: u8,
+    #[builder(default = 1)]
+    lines_between_queries: u8,
     /// Ignore case conversion for specified strings in the array.
-    ///
-    /// Default: None
-    pub ignore_case_convert: Option<Vec<&'a str>>,
+    ignore_case_convert: Option<Vec<&'a str>>,
```

Comment:
This could be later changed to accept AsRef<[&str]> so one can pass to the builder also `&[&str]`,`[&str; N]` and so on.